src/components/encoding-pane/filter-shelf.tsx

```diff
@@ -58,31 +68,128 @@ class FilterShelfBase extends React.Component<FilterShelfProps, {}> {
     });
   }
 
+  protected filterModifyTimeUnit(e: any, index: number, domain: number[]) {
+    const {handleAction} = this.props;
+    let timeUnit = e.target.value;
+    if (timeUnit === '-') {
+      timeUnit = undefined;
+    }
+
+    handleAction({
+      type: FILTER_MODIFY_TIME_UNIT,
+      payload: {
+        index,
+        timeUnit
+      }
+    });
+    if (!timeUnit) {
+      // reset range
+      handleAction({
+        type: FILTER_MODIFY_EXTENT,
+        payload: {
+          index,
+          range: domain
+        }
+      });
+      return;
+    } else if (timeUnit === TimeUnit.MONTH || timeUnit === TimeUnit.DAY) {
+      handleAction({
+        type: FILTER_MODIFY_ONE_OF,
+        payload: {
+          index,
+          oneOf: getRange(domain, timeUnit)
+        }
+      });
+    } else {
+      handleAction({
+        type: FILTER_MODIFY_EXTENT,
+        payload: {
+          index,
+          range: getRange(domain, timeUnit)
+        }
+      });
+    }
+  }
+
   private renderFilterShelf(filter: RangeFilter | OneOfFilter, index: number) {
-    const {fieldDefs, schema} = this.props;
+    const {fieldDefs, schema, handleAction} = this.props;
     const fieldIndex = schema.fieldNames().indexOf(filter.field);
     const fieldDef = fieldDefs[fieldIndex];
-    const domain = schema.domain(fieldDef as FieldQuery);
-    return (
-      <div styleName='filter-shelf' key={index}>
-        <div styleName='header'>
-          <span>{filter.field}</span>
-          <a onClick={this.filterRemove.bind(this, index)}>
-            <i className='fa fa-times'/>
-          </a>
-        </div>
-        {this.renderFilter(filter, index, domain, fieldDef.type)}
-      </div>
+    let domain = schema.domain(fieldDef as FieldQuery);
+    const popupComponent = (
+      fieldDef.type === ExpandedType.TEMPORAL ?
+        this.renderTimeUnitChanger(filter, index, domain) :
+      null
     );
-  }
 
-  private renderFilter(filter: RangeFilter | OneOfFilter, index: number, domain: any[], type: ExpandedType) {
-    const {handleAction} = this.props;
+    let filterComponent;
     if (isRangeFilter(filter)) {
-      return <RangeFilterShelf domain={domain} index={index} filter={filter} handleAction={handleAction} type={type}/>;
+      const timeUnit = filter.timeUnit;
+      if (timeUnit) {
+        domain = getRange(domain, timeUnit);
+        if (timeUnit === TimeUnit.MONTH || timeUnit === TimeUnit.DAY) {
+          filterComponent = (
+            <OneOfFilterShelf
```

Comment:
Why is `OneOfFilterShelf` the filter component  inside `if(isRangeFilter(filter)` ?